.github/actions/spelling/patterns.txt

```diff
@@ -20,5 +20,11 @@ sha[0-9]{3}-[0-9a-zA-Z+/]{64}
 # Hexadecimal literals in C# code.
 \b0x[0-9a-fA-F]+\b
 
+# Match the "I" from the beginning of interface names and the "T" from the beginning of generic type parameter names.
+# The pattern matches, if the first letter is "I" or "T", followed a capital letter, followed by a word character that
+# is not a capital letter, followed by 0 or more word characters. Only the first letter is captured, so the rest is
+# still spell-checked as they are usually made up of actual words.
+\b(I|T)(?=[A-Z][a-z0-9_][\w]*)
```

Comment:
Why would you allow digits after the capital letter?

IA7bla
TB3foo

?

Reply:
For our purposes, something stricter would probably be simpler since this is about interface and class names:
```suggestion
\b(I|T)(?=[A-Z][a-z]\w*)
```

Reply:
.NET/C# conventions allow numbers and some special characters in interface/class names. It's rare for us but can happen with an external library that we reference in our code. I'll remove the brackets around `\w`!